docs/architecture/deep-dives/authentication/index.md

```diff
@@ -0,0 +1,325 @@
+
+For API token requests, the following additional properties are added to the response:
+
+- `ApiUseKeyConnector` - set to `true` if the user is set to use Key Connector for master password
+  retrieval.
+
+## Step 3: Client handles the response
+
+When a token request is verified and sent back to the client, the client is responsible for storing
+the resulting data in state properly, so that the user can authenticate and their vault can be
+decrypted.
+
+The core logic for handling the token responses is in the `processTokenResponse()` method in our
+base
+[`LoginStrategy`](https://github.com/bitwarden/clients/blob/master/libs/common/src/auth/login-strategies/login.strategy.ts).
+There, we do the following:
+
+### Use the Authentication Information to Set up the User’s Account
+
+The response contains a JWT `access_token` that contains user claims and that can be provided on
+subsequent API requests to authenticate the user, as well as a `refresh_token` that can be used to
+request a new `access_token` without prompting the user to authenticate again.
+
+The claims on the `access_token` are used to initialize the user's `AccountProfile` in state, and
+the `access_token` and `refresh_token` are stored on the `AccountTokens` in state for use on
+subsequent API requests.
+
+### Use the Decryption Information to Prepare for Decryption
+
+The `AccountKeys` (stored in `account.keys` in state) are also set up in the user's account using
+the information provided in the token response.
+
+The important keys that are set at this point are:
+
+#### Device Key (`deviceKey`)
+
+The Device Key is the symmetric encryption key stored on the user's trusted device and is used to
+decrypt the user's symmetric encryption key when using Trusted Device Encryption.
+
+It is not returned on the request (as it never leaves the device), but it is set on the
+authenticated user's account state at this point.
+
+#### User Key (`userKey`)
+
+The User Key is the symmetric encryption key used to decrypt the user's vault.
+
+The specifics of how the User Key is set are defined in each login strategy:
+
+- **Password and Login with Device strategies**
+  - The `Key` in the token response is stored in `userKeyMasterKey` in the account state, and the
+    `masterKey` already in state from the user entering their master password is used to decrypt and
+    store the `userKey`.
+- **API strategy**
+  - The `Key` in the token response is stored in `userKeyMasterKey` in the account state.
+  - If the `ApiUseKeyConnector` property is set on the token response, we retrieve the `masterKey`
+    from state and use it to decrypt and store the `userKey`.
+- **SSO strategy**
+  - Trusted Device Encryption presents us with multiple ways to obtain the User Key, and these are
+    handled here, in the SSO login strategy.
+  - These include:
+    - Using the device key if the user is on a Trusted Device
+    - Using an approved Admin Approval request to receive the key
```

Comment:
Should we also include the login with device flow here?

Reply:
My thinking here was only documenting the scenarios that are handled in the strategy itself, which I think just includes the cases where the key is already there, or can be obtained automatically from the approved Admin Approval.  I felt that calling out the decryption options was diverting into TDE documentation too much.

Reply:
Oh, great point. I was reading too quickly and didn't realize that was scoped to the SSO strategy! Sorry!